Common/Systems/VanillaModifications/BossItemRemovals/RemovePowerCellFunctionality.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Mono.Cecil.Cil;
+using MonoMod.Cil;
+using Terraria.ID;
+
+namespace PathOfTerraria.Common.Systems.VanillaModifications.BossItemRemovals;
+
+internal class RemovePowerCellFunctionality : ModSystem
+{
+	public override void Load()
+	{
+		IL_Player.TileInteractionsUse += StopNaturalGolemSpawn;
+	}
+
+	private void StopNaturalGolemSpawn(ILContext il)
+	{
+		ILCursor c = new(il);
+
+		// Match Golem's NPC ID
+		if (!c.TryGotoNext(x => x.MatchLdcI4(NPCID.Golem)))
+		{
+			return;
+		}
+
+		ILLabel label = null;
+
+		// Match & get the label so we can exit early
+		if (!c.TryGotoNext(MoveType.After, x => x.MatchBrtrue(out label)) || label is null)
+		{
+			return;
+		}
+
+		c.Emit(OpCodes.Br, label);
```

Comment:
Don't use `Try` short-circuits unless you are in a loop, as doing so will delay discovery of patch breakages, which result in incorrect mod behavior.
```suggestion
		ILCursor c = new(il);

		// Match Golem's NPC ID
		c.GotoNext(x => x.MatchLdcI4(NPCID.Golem));

		// Match & get the label so we can exit early
		ILLabel label = null;
		c.GotoNext(MoveType.After, x => x.MatchBrtrue(out label));

		c.Emit(OpCodes.Br, label);
```